SDL/MacOSBarItems.mm

```diff
@@ -0,0 +1,326 @@
+//
+//  MacOSBarItems.mm
+//  PPSSPP
+//
+//  Created by Serena on 06/02/2023.
+//
+
+#include "Common/System/System.h"
+#include "Common/System/NativeApp.h"
+#include "Core/Config.h"
+#include "Common/Data/Text/I18n.h"
+#include "Common/StringUtils.h"
+#import <AppKit/AppKit.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+// NSMenuItem requires the use of an objective-c selector (aka the devil's greatest trick)
+// So we have to make this class
+@interface BarItemsManager : NSObject
++(instancetype)sharedInstance;
+-(void)setupAppBarItems;
+@property (assign) NSMenu *openMenu;
+@property (assign) std::shared_ptr<I18NCategory> mainSettingsLocalization;
+@end
+
+void initBarItemsForApp() {
+    [[BarItemsManager sharedInstance] setupAppBarItems];
+}
+
+// im soooooo sorry for whoever had to read this impl
```

Comment:
unnecessary comment :)

Reply:
This code is not nearly bad enough to apologize for, heh.

-[Unknown]